tests/virtiofs/datavolume.go

```diff
@@ -335,5 +337,116 @@ var _ = Describe("[sig-storage] virtiofs", decorators.SigStorage, func() {
+				libdv.WithNamespace(testsuite.NamespaceTestDefault),
+			)
+
+			By("creating the DataVolume")
+			dataVolume, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(testsuite.NamespaceTestDefault).Create(context.Background(), dataVolume, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("waiting until the DataVolume is ready")
+			if libstorage.IsStorageClassBindingModeWaitForFirstConsumer(libstorage.Config.StorageRWOFileSystem) {
+				Eventually(ThisDV(dataVolume), 30).Should(WaitForFirstConsumer())
+			}
+
+			vmi = libvmifact.NewFedora(
+				libvmi.WithFilesystemDV(dataVolume.Name),
+				libvmi.WithNamespace(testsuite.NamespaceTestDefault),
+			)
+
+			vmi, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			// with WFFC the run actually starts the import and then runs VM, so the timeout has to include both import and start
+			By("waiting until the DataVolume is ready")
+			libstorage.EventuallyDV(dataVolume, 500, HaveSucceeded())
+
+			By("waiting until the VirtualMachineInstance starts")
+			vmi = libwait.WaitForVMIPhase(vmi, []v1.VirtualMachineInstancePhase{v1.Running}, libwait.WithTimeout(500))
+
+			By("getting kubevirt client")
+			virtCli := kubevirt.Client()
+
+			By("creating a privileged pod")
+			podName := "info-pod"
+			debugPod := &k8sv1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: podName,
+				},
+				Spec: k8sv1.PodSpec{
+					Containers: []k8sv1.Container{
+						{
+							Name:    "informer",
+							Image:   "busybox",
+							Command: []string{"/bin/sh", "-c", "sleep 3600"},
+							Stdin:   true,
+							TTY:     true,
+							SecurityContext: &k8sv1.SecurityContext{
+								Privileged: ptr.To(true),
+							},
+						},
+					},
+					HostNetwork: true,
+					HostPID:     true,
+					HostIPC:     true,
+					NodeName:    vmi.Status.NodeName,
+				},
+			}
+			pod, err := virtCli.CoreV1().Pods(testsuite.NamespacePrivileged).Create(context.TODO(), debugPod, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("waiting until the pod is running")
+			Eventually(func() k8sv1.PodPhase {
+				pod, err := virtCli.CoreV1().Pods(testsuite.NamespacePrivileged).Get(context.TODO(), podName, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				return pod.Status.Phase
+			}, 30*time.Second, 1*time.Second).Should(Equal(k8sv1.PodRunning))
+
+			By("extracting process cgroup and NS info")
+			script := "export PID1=$(pidof virtiofs-placeholder) " +
+				"; grep ^0 /proc/$PID1/cgroup " +
+				"&& echo $(cd /proc/$PID1/ns && echo net pid ipc mnt cgroup uts | xargs -n1 readlink) " +
+				"&& export PID2=$(pidof virtiofsd) " +
+				"; grep ^0 /proc/$PID2/cgroup " +
+				"&& echo $(cd /proc/$PID2/ns && echo net pid ipc mnt cgroup uts | xargs -n1 readlink)" +
+				"&& awk -e '/^PPid:/ { print ENVIRON[\"PID1\"] == $2 }' /proc/$PID2/task/$PID2/status"
```

Comment:
Instead of executing this complex commands, can we simply read the file content and do some filetring directly in golang?

Reply:
Simplified it a bit, since I can use bash inside the virt-handler container